### PR TITLE
Fe/refactor  _transaction

### PR DESCRIPTION
**Pull request by bigboydiamonds**
## Summary by CodeRabbit

- **New Features**
	- Introduced a `TimeRemaining` component to display time-related information contextually.
	- Implemented a `TransactionStatus` component to clearly display transaction states.

- **Bug Fixes**
	- Improved transaction status and time calculations for more accurate information display.
	- Enhanced the transaction list's time update and sorting mechanism for better user experience.

- **Refactor**
	- Streamlined transaction-related components by reorganizing and refactoring code for better maintainability.
	- Replaced direct hook usage with custom `useBridgeTxUpdater` and `useIntervalTimer` hooks for optimized performance.

- **Documentation**
	- No visible documentation updates for end-users.

- **Style**
	- No visible style updates for end-users.

- **Tests**
	- No visible test updates for end-users.

- **Chores**
	- Cleaned up unused imports and components to declutter the interface.

- **Revert**
	- No reverts in this release.

430ac1b445d2dcfa877d69b37079db6186a6b63e: [synapse-interface preview link ](https://sanguine-synapse-interface-329koqqu8-synapsecns.vercel.app)
20e250098bbee5b1909a2bce5179899929c02f45: [synapse-interface preview link ](https://sanguine-synapse-interface-2nkothvku-synapsecns.vercel.app)
b3957aaae1f1f7418f60a6f4eb55997979c44670: [synapse-interface preview link ](https://sanguine-synapse-interface-8almlpinn-synapsecns.vercel.app)
285f4e390ac5438a3e0402c0faf667286847dbf3: [synapse-interface preview link ](https://sanguine-synapse-interface-fhfot5df3-synapsecns.vercel.app)
c9ba3d8df396d52689f841f1524568500ee4a60f: [synapse-interface preview link ](https://sanguine-synapse-interface-n8jasrqto-synapsecns.vercel.app)
76c7d6e7d9f6783d0b39303418917226f72c40c3: [synapse-interface preview link ](https://sanguine-synapse-interface-3a1sr0ojn-synapsecns.vercel.app)
b1d9549755c65092811b545bfc943884791f39f4: [synapse-interface preview link ](https://sanguine-synapse-interface-5do0sb98g-synapsecns.vercel.app)
901e82c315e7f6d42a5d6034bfc5fd924e0bcfd1: [synapse-interface preview link ](https://sanguine-synapse-interface-fgcgoth8r-synapsecns.vercel.app)
a4b82872c3a755d9ef8551fb1de8b0f35b6220fb: [synapse-interface preview link ](https://sanguine-synapse-interface-c80lt873q-synapsecns.vercel.app)
c3ad0c77fcd3d06913cbeb13bb6479758adb26fb: [synapse-interface preview link ](https://sanguine-synapse-interface-5j9y2rdi0-synapsecns.vercel.app)
487bb584ce8b0c8355359b63aa4c2dd39615b1c0: [synapse-interface preview link ](https://sanguine-synapse-interface-eeo5ec0d3-synapsecns.vercel.app)
2bf6af689e82cafc6b283a330f6136240224cd39: [synapse-interface preview link ](https://sanguine-synapse-interface-5kqekvlph-synapsecns.vercel.app)
59fe0b2874f082d2f94d6205f4dbec0db6e4f69c: [synapse-interface preview link ](https://sanguine-synapse-interface-adl39thk8-synapsecns.vercel.app)

### CODE REVIEW

File: packages/synapse-interface/components/_Transaction/components/TimeRemaining.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  isComplete: boolean
+  remainingTime: number
+  isDelayed: boolean
+}) => {
+  if (isComplete) return
+
+  if (isDelayed) {
+    return <div>Waiting...</div>
+  }
+
```
Comment (abtestingalpha) [inline packages/synapse-interface/components/_Transaction/components/TimeRemaining.tsx:15]: In general, there's usage of ms vs seconds vs minutes in the app, and think we should stick to one (prob ms/seconds) and adjust where we need to. Goal is to be consistent so when we add more around all this, there's no confusion about what unit we're dealing with.

Here, for example, `remainingTime` is minutes though some txns will be < than a minute. We can do something like this and use `estTime` on line 16.

```
const estTime = useMemo(() => {
    if (remainingTime > 60) {
      return Math.ceil(remainingTime / 60) + ' minutes'
    } else {
      return remainingTime + ' seconds'
    }
  }, [remainingTime])
```

File: packages/synapse-interface/components/_Transaction/helpers/getEstimatedTimeStatus.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  const remainingTimeInMinutes = Math.ceil(remainingTime / oneMinuteInSeconds)
+  const elapsedTimeInMinutes = Math.ceil(elapsedTime / oneMinuteInSeconds)
+
+  const isEstimatedTimeReached = remainingTime < 0
+  const isStartCheckingTimeReached = remainingTime < oneMinuteInSeconds
+
+  return {
+    elapsedTime,
+    elapsedTimeInMinutes,
```
Comment (abtestingalpha) [inline packages/synapse-interface/components/_Transaction/helpers/getEstimatedTimeStatus.ts:26]: Similar comment about time before. Additionally `Math.ceil` truncates a value (presumably to show buffer), but we don't want that if we're using it to make comparisons. Would remove all the `inMinutes` fields from this hook) and just use seconds and convert to minutes at the end of where they're used.

**Review by abtestingalpha — CHANGES_REQUESTED** [review]

File: packages/synapse-interface/components/_Transaction/components/TimeRemaining.tsx (hunk 1 above)
Comment (bigboydiamonds) [reply packages/synapse-interface/components/_Transaction/components/TimeRemaining.tsx:24]: Updated here: https://github.com/synapsecns/sanguine/pull/1892/commits/770029f354ee48eaa6e4ef4fca0670ea5eaa328f

**Review by abtestingalpha — CHANGES_REQUESTED** [review]
Seeing a regression here where after bridge completes, destination chain balance doesn't update (currently working on master):

Screen Shot 2024-01-23 at 3 34 32 PM

**Comment by bigboydiamonds** [comment]
> Seeing a regression here where after bridge completes, destination chain balance doesn't update (currently working on master). USDC for example should be populated (waited a while to see if it was slow but didn't udpate):
> 
> Screen Shot 2024-01-23 at 3 34 32 PM

Fixed in https://github.com/synapsecns/sanguine/pull/1892/commits/87f970beda90d480f01d6c6be15a833f57f89634

**Review by abtestingalpha — APPROVED** [review]

**Pull request merged**